webapp/src/dogma/common/components/DataTable.tsx

```diff
@@ -0,0 +1,95 @@
+import { Table, Thead, Tbody, Tr, Th, Td, chakra, Text } from '@chakra-ui/react';
+import { TriangleDownIcon, TriangleUpIcon } from '@chakra-ui/icons';
+import {
+  useReactTable,
+  flexRender,
+  getCoreRowModel,
+  ColumnDef,
+  SortingState,
+  getSortedRowModel,
+  ColumnFiltersState,
+  getFilteredRowModel,
+} from '@tanstack/react-table';
+import { Filter } from './Filter';
+import { useState } from 'react';
+
+export type DataTableProps<Data extends object> = {
+  data: Data[];
+  columns: ColumnDef<Data, any>[];
+  handleOnClick: Function;
+};
+
+export const DataTable = <Data extends object>({ data, columns, handleOnClick }: DataTableProps<Data>) => {
+  const [sorting, setSorting] = useState<SortingState>([]);
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+  const table = useReactTable({
+    columns,
+    data,
+    getCoreRowModel: getCoreRowModel(),
+    onSortingChange: setSorting,
+    getSortedRowModel: getSortedRowModel(),
+    onColumnFiltersChange: setColumnFilters,
+    getFilteredRowModel: getFilteredRowModel(),
+    state: {
+      sorting,
+      columnFilters,
+    },
+  });
+
+  return (
+    <>
+      <Text mb="8px">Filter by {table.getHeaderGroups()[0].headers[0].id /* Filter by the 1st column */} </Text>
```

Comment:
It would take some time to learn `react-table` and understand this code.

Reply:
This one is from ChakraUI's example https://chakra-ui.com/getting-started/with-react-table